src/robot/running/namespace.py

```diff
@@ -338,6 +339,42 @@ def _get_runner_from_libraries(self, name):
             return found[0]
         self._raise_multiple_keywords_found(name, found)
 
+    def _handle_private_runners(self, runners):
+        sum_public = sum("robot:private" not in runner.tags for runner in runners)
+        sum_private = sum("robot:private" in runner.tags for runner in runners)
+
+        if sum_public == len(runners) or sum_private == len(runners) or sum_public > 1:
+            return runners
+
+        for runner in runners:
+            if 'robot:private' not in runner.tags:
+                public_runner = runner
+                private_runners = runners.copy()
+                private_runners.remove(public_runner)
+                break
+
+        self._public_and_private_keyword_conflict_warning(public_runner, private_runners)
+        return [public_runner]
+
+    def _public_and_private_keyword_conflict_warning(self, public_runner, private_runners):
+        private_runners_str = ""
+        for runner in private_runners:
+            if runner != private_runners[-1]:
+                private_runners_str += f"'{runner.longname}' / "
+            else:
+                private_runners_str += f"'{runner.longname}'"
```

Comment:
This looks a bit complicated way to concatenate names with ` /  `. Just using something like
```python
private_runner_str = ' / '.join(r.longname for r in private_runners)
```
ought to handler it as a one-liner. I'm also not certain is using ` / ` as a separator between private keywords here that good in general. Just listing _all_ runners like `'First', 'Second' or 'Third` would be simpler to implement and would look, in my opinion, clearer. I can test that in practice myself.